src/charm.py

```diff
@@ -318,6 +320,8 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
                 return
             if self.unit.is_leader():
                 self._store_certificates_from_config()
+                self.tls_certificates.revoke_all_certificates()
+                logger.info("Revoked all previously issued certificates.")
```

Comment:
Another thought, now may be late for this specific PR but the logging for revoking certs should probably be in the lib.